tests/test_set_mod_state.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+import torch
+from parameterized import parameterized
+
+from monai.networks.utils import compatible_mod_state
+from monai.utils import set_determinism
+
+
+class _TestModelOne(torch.nn.Module):
+    def __init__(self, n_n, n_m, n_class):
+        super(_TestModelOne, self).__init__()
+        self.layer = torch.nn.Linear(n_n, n_m)
+        self.class_layer = torch.nn.Linear(n_m, n_class)
+
+    def forward(self, x):
+        x = self.layer(x)
+        x = self.class_layer(x)
+        return x
+
+
+class _TestModelTwo(torch.nn.Module):
+    def __init__(self, n_n, n_m, n_d, n_class):
+        super(_TestModelTwo, self).__init__()
+        self.layer = torch.nn.Linear(n_n, n_m)
+        self.layer_1 = torch.nn.Linear(n_m, n_d)
+        self.class_layer = torch.nn.Linear(n_d, n_class)
+
+    def forward(self, x):
+        x = self.layer(x)
+        x = self.layer_1(x)
+        x = self.class_layer(x)
+        return x
+
+
+TEST_CASES = []
+__devices = ("cpu", "cuda") if torch.cuda.is_available() else ("cpu",)
+for _x in __devices:
+    for _y in __devices:
+        TEST_CASES.append((_x, _y))
+
+
+class TestModuleState(unittest.TestCase):
+    def tearDown(self):
+        set_determinism(None)
+
+    @parameterized.expand(TEST_CASES)
+    def test_set_state(self, device_0, device_1):
+        set_determinism(0)
+        model_one = _TestModelOne(10, 20, 3)
+        model_two = _TestModelTwo(10, 20, 10, 4)
+        model_one.to(device_0)
+        model_two.to(device_1)
+        model_dict, ch, unch = compatible_mod_state(model_one, model_two)
+        model_one.load_state_dict(model_dict)
```

Comment:
Could the API be changed so that these two are done in the same function? Is there an alternative utility for `model_dict`, or is it just there to then be loaded into a model?

i.e., have `compatible_mod_state` call `load_state_dict` such that:

```python
model_one, ch, unch = compatible_mod_state(model_one, model_two)
```

Reply:
ok I can add an `inplace=True` option